_go/postgres/pgx.go

```diff
@@ -0,0 +1,369 @@
+package postgres
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"log"
+	"regexp"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/edgedb/webapp-bench/_go/bench"
+	"github.com/edgedb/webapp-bench/_go/cli"
+	"github.com/jackc/pgx/v4/pgxpool"
+)
+
+func PGXWorker(args cli.Args) (bench.Exec, bench.Close) {
+	url := fmt.Sprintf(
+		"postgresql://postgres_bench:edgedbbenchmark@%v:%v/postgres_bench",
+		args.Host,
+		args.Port,
+	)
+
+	pool, err := pgxpool.Connect(context.TODO(), url)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	regex := regexp.MustCompile(`users|movie|person`)
+	queryType := regex.FindString(args.Query)
+
+	var exec bench.Exec
+	switch queryType {
+	case "movie":
+		exec = pgxExecMovie(pool, args)
+	case "person":
+		exec = pgxExecPerson(pool, args)
+	case "users":
+		exec = pgxExecUser(pool, args)
+	default:
+		log.Fatalf("unknown query type: %q", queryType)
+	}
+
+	return exec, pool.Close
+}
+
+func pgxExecMovie(pool *pgxpool.Pool, args cli.Args) bench.Exec {
+	var (
+		movie Movie
+		wg    sync.WaitGroup
+	)
+
+	queries := strings.Split(args.Query, ";")
+	m := pgxGetMovie(pool, queries[0], &movie)
```

Comment:
This isn't exactly equivalent to the other benchmarks, where we reconstruct the JSON-like nested structure from multiple queries.  For it to be a fair benchmark, we must do that here also.  I don't expect a huge change, but it might still be a factor.

Reply:
What do you mean by `JSON-like nested structure`? The Movie type is a nested structure and movie is populated from multiple queries?

Reply:
Actually, ignore me here.  The returned structure is mostly equivalent.